mdagent/tools/base_tools/analysis_tools/secondary_structure.py

```diff
@@ -0,0 +1,551 @@
+    description = """Compute the relative shape antisotropy for each
+    frame in a molecular dynamics trajectory. Input is a trajectory
+    file ID and an optional topology file ID.
+    The output is an array of relative shape antisotropy values
+    for each frame of the trajectory."""
+    path_registry: PathRegistry = PathRegistry.get_instance()
+
+    def __init__(self, path_registry: PathRegistry):
+        super().__init__()
+        self.path_registry = path_registry
+
+    def _compute_relative_shape_antisotropy(self, traj: md.Trajectory) -> np.ndarray:
+        """
+        Computes the relative shape antisotropy for a trajectory.
+
+        Args:
+            traj: The trajectory to compute the relative shape antisotropy for.
+
+        Returns:
+            An array of relative shape antisotropy
+                values for each frame of the trajectory.
+        """
+        return md.relative_shape_antisotropy(traj)
+
+    def _run(self, traj_file: str, top_file: str | None = None) -> str:
+        """
+        Computes the relative shape antisotropy for a trajectory and saves the results
+        to a file.
+
+        Args:
+            traj_file: The file id of the trajectory to
+                compute the relative shape antisotropy for.
+            top_file: The file id of the topology file to use.
+
+        Returns:
+            A message indicating the success of the computation.
+        """
+        try:
+            traj = load_single_traj(
+                path_registry=self.path_registry,
+                traj_fileid=traj_file,
+                top_fileid=top_file,
+            )
+            if not traj:
+                raise Exception("Trajectory could not be loaded.")
+        except Exception as e:
+            return str(e)
+        relative_shape_antisotropy_values = self._compute_relative_shape_antisotropy(
+            traj
+        )
+        if traj.n_frames == 1:
+            return (
+                "Relative shape antisotropy computed for "
+                "a single frame, no file saved."
+                f"Relative shape antisotropy: {relative_shape_antisotropy_values}"
+            )
+
+        raw_file_id = write_raw_x(
+            "relative_shape_antisotropy",
+            relative_shape_antisotropy_values,
+            traj_file,
+            self.path_registry,
+        )
+        plot_file_id = plot_x_over_time(
+            "relative_shape_antisotropy",
+            relative_shape_antisotropy_values,
+            traj_file,
+            self.path_registry,
+        )
+        return (
+            "relative_shape_antisotropy_values saved to "
+            f"{raw_file_id}, plot saved to "
+            f"{plot_file_id}"
+        )
+
+    async def _arun(self, traj_file, top_file):
+        """Runs the tool asynchronously."""
+        raise NotImplementedError("Async version not implemented")
+
+
+class AnalyzeProteinStructure(BaseTool):
+    name = "AnalyzeProtein"
+    description = (
```

Comment:
cause as of now, if instead of n_chains, the agent can use for example 'total chains', if he actually knows how haha, as it doesn't say what he can ask for.

Reply:
fixed - I changed the fan to always give a count of all